### ISSUE TITLE
Lost some features columns when using feature importance

### ISSUE THREAD

**Issue by Daemonyz** [dmlc/xgboost#4324]
When i use model.get_score() or xgb.plot_importance(), only 6 features importance turn out, while i feed 8 features totally. And I set the max_num_features to a large number.

So in what conditions xgboost will drop or skip some feature columns?

**Comment by hcho3** [comment]
XGBoost will use whatever set of split features to minimize the training loss, so there is no guarantee that all features would be used. Try using feature interaction constraints if you’d like to enforce all fea  To be used.
